bin/setup.sh

```diff
@@ -6,27 +6,53 @@
 #
 set -euxo pipefail
 
-# NOTE: this is written under the assumption that it will be built in canon
-sudo apt -y update && sudo apt -y upgrade && sudo apt install -y cmake python3.11 python3.11-venv wget
+if [[ ${OS} == "darwin" ]]; then
+    echo "Detected MacOS ${ARCH}"
+
+  if ! command -v brew >/dev/null; then
+      echo "Installing Homebrew"
+      /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  fi
+  # Install required tools
+  brew install cmake python@3.11 wget unzip conan || true
+elif  [[ ${OS} == "linux" ]]; then
+    echo "Detected Linux $ARCH"
+    # NOTE: this is written under the assumption that it will be built in canon
+    sudo apt -y update && sudo apt -y upgrade && sudo apt install -y cmake python3.11 python3.11-venv wget
+else
+    echo "Unsupported OS: ${OS}"
+    exit 1
+fi
```

Comment:
I'm a bit concerned about this if this is supposed to run in a non-isolated environment on my mac. Installing brew for the user even if it's just for dev users feels a bit scary. Do we have ways to make this less system-breaky?

Reply:
I'm gonna open up a thread in #team-devops about how we should best handle this https://viaminc.slack.com/archives/C02P30REXFW/p1751560279076289

Reply:
I don't think we have a containerization option on macos at the moment. 

I think the few things we are installing are minimally invasive. 

I think we can work with this as a v1.